From 08eb485a006dc2baca42eb280ae60e9f0fc00dff Mon Sep 17 00:00:00 2001
From: Alexey Edelev <alexey.edelev@qt.io>
Date: Mon, 23 Nov 2020 18:05:24 +0100
Subject: [PATCH] CMake: Change generated resources naming

Remove redundant 'target' prefix for generated resource name for qml
files. It's expected that 'qt6_add_resources' will created resource
targets prefixed with '${target}_'.

Fixes: QTBUG-88581
Pick-to: 6.0
Change-Id: Ie63fce11931a2cb566aa1902a77227ef86ca1ea8
Reviewed-by: Alexandru Croitor <alexandru.croitor@qt.io>
---

diff --git a/src/qml/Qt6QmlMacros.cmake b/src/qml/Qt6QmlMacros.cmake
index 7cf2e9b..9d815d3 100644
--- a/src/qml/Qt6QmlMacros.cmake
+++ b/src/qml/Qt6QmlMacros.cmake
@@ -389,9 +389,7 @@
             PROPERTIES QT_RESOURCE_ALIAS "qmldir"
         )
 
-        set(resource_target "Foo")
         qt6_add_resources(${target} ${qmldir_resource_name}
-            PREFIX ${target_resource_prefix}
             FILES "${qmldir_file}"
             OUTPUT_TARGETS resource_targets
         )
@@ -585,8 +583,7 @@
     math(EXPR new_count "${resource_count} + 1")
     set_target_properties(${target} PROPERTIES QT6_QML_MODULE_ADD_QML_FILES_COUNT ${new_count})
 
-    # TODO: Fix this if it blows up on Windows due to too long target name.
-    qt6_add_resources(${target} "${target}_qml_files_resource_${new_count}"
+    qt6_add_resources(${target} "qml_files_${new_count}"
         FILES ${arg_FILES}
         OUTPUT_TARGETS resource_targets
     )
